### PR TITLE
[6.3 Feature] Ui csv test stubs

### PR DESCRIPTION
**Pull request by pgagne**

### CODE REVIEW

File: tests/foreman/ui/test_ui_csv_export.py [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+:CaseAutomation: notautomated
+
+:Upstream: No
+"""
+
+from robottelo.decorators import stubbed, tier1
+from robottelo.test import UITestCase
+
+
+class UICSVExportHosts(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: A it is already on `ui` no need the `UI` prefix, call it just `CSVExportHostTestCase` dont forget the `TestCase` suffix.

File: tests/foreman/ui/test_ui_csv_export.py [hunk 2]
```diff
@@ -0,0 +1,732 @@
...
+
+    @stubbed
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: It takes  parentesis `@stubbed()`

File: tests/foreman/ui/test_ui_csv_export.py (hunk 1 above)
Comment (pgagne) [reply tests/foreman/ui/test_ui_csv_export.py]: The UI here is part of the feature name, to distinguish it from the CLI CSV export

**Comment by rochacbruno** [comment]
Filename can also be changed from `tests/foreman/ui/test_ui_csv_export.py` to `tests/foreman/ui/test_csv_export.py`. No need to repeat `ui` in the name

File: tests/foreman/ui/test_ui_csv_export.py [hunk 3]
```diff
@@ -0,0 +1,732 @@
...
+:CaseAutomation: notautomated
+
+:Upstream: No
+"""
+
+from robottelo.decorators import stubbed, tier1
+from robottelo.test import UITestCase
+
+
+class CSVExportHosts(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: `CSVExportHostsTestCase`

File: tests/foreman/ui/test_ui_csv_export.py [hunk 4]
```diff
@@ -0,0 +1,732 @@
...
+            Populate host page with entries with no last report time
+        :steps:
+            1. Export CSV page in the UI
+
+        :expectedresults:
+            Last reported time is in the correct format in the CSV.
+        """
+
+
+class UICSVExportFacts(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: `CSVExportFactsTestCase`

File: tests/foreman/ui/test_ui_csv_export.py [hunk 5]
```diff
@@ -0,0 +1,732 @@
...
+        :id: 558eb900-c39e-4f8a-a816-4b935ca4aa17
+
+        :steps:
+            1. Verify what ever the end result of
+               https://bugzilla.redhat.com/show_bug.cgi?id=1477341 is.
+
+        """
+
+
+class UICSVExportFactsNested(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: `CSVExportFactsNestedTestCase`

File: tests/foreman/ui/test_ui_csv_export.py [hunk 6]
```diff
@@ -0,0 +1,732 @@
...
+
+        :steps:
+            1. Recurse down a nested fact (ex DMI)
+            2. Export CSV for each level
+
+        :expectedresults: Values are in CSV at each level.
+        """
+
+
+class UICSVExportConfigMgmtReports(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: `CSVExportConfigMgmtReportsTestCase`

File: tests/foreman/ui/test_ui_csv_export.py [hunk 7]
```diff
@@ -0,0 +1,732 @@
...
+
+    @stubbed()
+    @tier1
+    def test_positive_single_page(self):
+        """
+        :id: 9006c4fa-1e46-4f9f-bb22-5b5fb96ba94b
+
+        :step
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: what is the `:steps:` here?

File: tests/foreman/ui/test_ui_csv_export.py [hunk 8]
```diff
@@ -0,0 +1,732 @@
...
+               filterA and some dont
+        :steps:
+            1. Filter the host page with filterA
+            2. Export the host page to CSV
+
+        :expectedresults: CSV contains filtered hosts from page.
+        """
+
+
+class CSVExportConfigMgmtColumns(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_ui_csv_export.py]: `CSVExportConfigMgmtColumnsTestCase`

**Comment by pgagne** [comment]
@rochacbruno fixed the file name

File: tests/foreman/ui/test_csv_export.py [hunk 9]
```diff
@@ -0,0 +1,732 @@
...
+:CaseAutomation: notautomated
+
+:Upstream: No
+"""
+
+from robottelo.decorators import stubbed, tier1
+from robottelo.test import UITestCase
+
+
+class CSVExportHosts(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_csv_export.py]: `CSVExportHostsTestCase`

File: tests/foreman/ui/test_csv_export.py [hunk 10]
```diff
@@ -0,0 +1,732 @@
...
+               some match filterA and some dont
+        :steps:
+            1. Filter the host page with filterA
+            2. Export the host page to CSV
+
+        :expectedresults: CSV contains filtered hosts from page.
+        """
+
+
+class UICSVExportHostsColumns(UITestCase):
```
Comment (rochacbruno) [inline tests/foreman/ui/test_csv_export.py]: `CSVExportHostsColumnsTestCase`

and aplly to all others

File: tests/foreman/ui/test_csv_export.py [hunk 11]
```diff
@@ -0,0 +1,732 @@
...
+
+class UICSVExportConfigMgmtReports(UITestCase):
+
+    @stubbed()
+    @tier1
+    def test_positive_single_page(self):
+        """
+        :id: 9006c4fa-1e46-4f9f-bb22-5b5fb96ba94b
+
+        :step
```
Comment (rochacbruno) [inline tests/foreman/ui/test_csv_export.py]: `:steps:` are missing

**Comment by sghai** [comment]
Please rebase once and see if you can sqash all commits.. there are so many commits. not sure where should I start w/.

**Comment by pgagne** [comment]
Can't you squash and merge it?

**Comment by pgagne** [comment]
Can't you squash and merge it using the GitHub feature?

**Comment by pgagne** [comment]
@sghai @rochacbruno @SatelliteQE/quality-engineers does this look good?

File: tests/foreman/ui/test_csv_export.py [hunk 12]
```diff
@@ -0,0 +1,781 @@
...
+from robottelo.decorators import stubbed, tier1
+from robottelo.test import UITestCase
+
+
+class CSVExportHostsTestCase(UITestCase):
+
+    @stubbed()
+    @tier1
+    def test_positive_single_page(self):
+        """
```
Comment (rochacbruno) [inline tests/foreman/ui/test_csv_export.py]: Usually we include a description of test as the beggining of the doctest as in example: https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_activationkey.py#L138

```py
def test_positive_export_with_single_page(self):
    """Assure the exported CSV has all rows when exporting a single page.
    :id: be5e2b2d-3a93-4803-8c6e-d81a97c5a1a5
    :steps:
     .....
    """
```

And looks like all other cases are missing that descriptions

File: tests/foreman/ui/test_csv_export.py [hunk 13]
```diff
@@ -0,0 +1,781 @@
...
+            Populate hosts multiple hosts with a mix of Puppet Enviromets,
+            Hostgroups, OS, etc
+
+        :steps:
+            1. Export the CSV Page in the UI
+
+        :expectedresults:
+            Exported CSV will contain text representations of all columns.
+
+
```
Comment (rochacbruno) [inline tests/foreman/ui/test_csv_export.py]: Too much blank lines at the end of docstring

Comment (pgagne) [reply tests/foreman/ui/test_csv_export.py]: @rochacbruno @sghai if you think its ok as is please merge it.

**Pull request merged**